src/core/Akka.Remote.Tests/EndpointRegistrySpec.cs

```diff
@@ -148,6 +148,71 @@ public void EndpointRegistry_should_be_able_to_register_Quarantined_policy_for_a
             Assert.Equal(42, reg.WritableEndpointWithPolicyFor(address1).AsInstanceOf<EndpointManager.Quarantined>().Uid);
             Assert.Equal(deadline, reg.WritableEndpointWithPolicyFor(address1).AsInstanceOf<EndpointManager.Quarantined>().Deadline);
         }
+
+        [Fact]
```

Comment:
Specs to cover the original case reported in #1845 and other cases where bad behavior could occur.